monai/data/image_writer.py

```diff
@@ -414,9 +415,12 @@ def set_data_array(
             spatial_ndim=kwargs.pop("spatial_ndim", 3),
             contiguous=kwargs.pop("contiguous", True),
         )
-        self.channel_dim = (
-            channel_dim if self.data_obj is not None and len(self.data_obj.shape) >= _r else None
-        )  # channel dim is at the end
+        if channel_dim is None:
+            self.channel_dim = -1  # after convert_to_channel_last the last dim is the channel
```

Comment:
I think there has a bug. Since the default value of `squeeze_end_dims` is `True` in `SaveImage`, the `data_obj` after `convert_to_channel_last` will always don't have a channel. 
https://github.com/Project-MONAI/MONAI/blob/678b5124a74f32859d692d142728ca1f7fa98a25/monai/transforms/io/array.py#L382

Reply:
hmm with auto squeeze_end_dims, we'll lose track of the channel dim